### DISCUSSION TITLE
Whats the reasoning behind the coding style in headers? (curiousity)

### DISCUSSION THREAD

**Discussion by Mahmoud1205** [jrouwe/JoltPhysics#1233] [General]
For example, DebugRenderer.h
```cpp
         // Why so much spacing between void and DrawLine?
virtual void						DrawLine(RVec3Arg inFrom, RVec3Arg inTo, ColorArg inColor) = 0;
```
I understand if it is to make each function name on the same column as other function names, but every function name is VERY far from its return type, why is that?
I'm not trying to come out as aggressive or say that this is a wrong or bad way of writing code, but I am just curious on why you write like this since maybe there's benefits to this way of writing that I don't know.

**Comment by jrouwe** [comment]
I think you can only understand the reason if you look at it in the context of a file:

![image](https://github.com/user-attachments/assets/28be2c4e-aa89-4b7f-b92b-fb3147f06a1d)

The function names of all the function in a class align so that it is easier to spot them. The amount of tabs vary from file to file and is determined by the longest return type / inline specifier. In the case of DebugRenderer.h there is indeed an excessive amount of tabs, so there must have been a really long return type in the past that doesn't exist anymore. I've cleaned it up.

This coding style is similar (but not exactly the same) as the coding style that is used at Guerrilla where I've worked for the past 24 years, so I just kept what I was used to.

**Reply by Mahmoud1205** [reply]
Okay, just one last question.

Just so I can make sure I understood you, are all the functions in DebugRenderer.h after you edited it aligned according to the `using SupportFunction ...` line?
![Screenshot 2024-08-25 221528](https://github.com/user-attachments/assets/e6355453-9d68-4191-808d-c1f5d6932791)
And what are the functions in Math/Vec3.h aligned according to? Because to me it just looks like they all have some extra spacing. Or do you just add extra spacing to them for extra readability?

Also thanks for making this great library open-source.

**Reply by jrouwe** [reply]
> Are these two one-space tabs? If so whats the point of them and why not just use one two-space tab?

Tabs are 4 spaces as you can see in the `.editorconfig` file and that Vec3 just happens to be 1 space away from the next tab stop.

> Just so I can make sure I understood you, are all the functions in DebugRenderer.h after you edited it aligned according to the using SupportFunction ... line?

The `using` lines don't count, it's only the return values or member types. In this case `SwingPyramidBatches` is longest. 

For Vec3.h it's this line:

![image](https://github.com/user-attachments/assets/b8678288-e2fd-4cd1-a9c6-7ee74b6604ee)

Technically it can lose one more tab but then there's only 1 space left, so I usually add another tab then.

I have to say it's not an exact science. As long as most function names align and they're clearly separated I'm happy.

**Reply by Mahmoud1205** [reply]
Okay, thanks
